### PR TITLE
Add SQA design documents for NS and ray tracing

### PR DESCRIPTION
**Pull request by GiudGiud**
@grmnptr @lindsayad @tanoret fyi

I think we can expand on what we know we are missing at this point

My limited knowledge of the work on the FE side is likely transparent in this PR.

### CODE REVIEW

File: modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md
Comment (lindsayad) [inline modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:22]:
```suggestion
It should be able to determine the pressure, velocity, and fluid temperature fields, as well
as the solid temperature field in the case of conjugate heat transfer simulations. It aims to also be able to
```

File: modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md
Comment (lindsayad) [inline modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:26]: I might delete this sentence. I think it is unnecessarily narrowing. In the very short term the module will also be the backbone of the SAM multi-D capability. Moreover, though the scope for NEAMS is coarse mesh, I am hopeful that if not now, then some day it can also be used for finer meshes as well, e.g. when we have scalable solvers

**Review by lindsayad — DISMISSED** [review]
Just reviewing the NS part. Very nice

File: modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md
Comment (GiudGiud) [reply modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:26]: We could update when the SAM multi-D is merged in.
Similarly once we have shown we can do finer meshes in some capacity we can update.

but actually are these documents updateable when needed @cticenhour ?

Comment (cticenhour) [reply modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:26]: These documents are living ones, meant to be kept up to date as the code they're associated with change. I tend toward more broad statements and refer back to design documentation for specifics myself, but I leave those sorts of things to the module maintainers for how they want to outline the contents of their code.

**Comment by GiudGiud** [comment]
Review should be addressed

Still need eyes on ray tracing. Logan and I wrote it together so need another reviewer

File: modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md
Comment (GiudGiud) [reply modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md]: removed. We ll update after SAM merge

File: modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_sdd.md
Comment (cticenhour) [inline modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_sdd.md:56]:
```suggestion
Each major class of discretizations has its own index page, notably the [continuous Galerkin finite element](navier_stokes/cgfe.md),
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Only one typo I found. Anymore work needed here @GiudGiud?

**Comment by GiudGiud** [comment]
nope, good to go as far as I know. Typo fix is squashed in

**Review by cticenhour — APPROVED** [review]

**Pull request merged**
